packages/core/src/common/resource.ts

```diff
@@ -14,8 +15,8 @@ import { MaybePromise } from "./types";
 
 export interface Resource extends Disposable {
     readonly uri: URI;
-    readContents(options?: { encoding?: string }): Promise<string>;
-    saveContents?(content: string, options?: { encoding?: string }): Promise<void>;
+    readContents(): Promise<string>;
+    saveContents?(content: string, changes?: TextDocumentContentChangeEvent[]): Promise<void>;
```

Comment:
This signature looks strange to me. I'd expect to either set the content or apply changes.